File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin() , strs.end());

        int n = strs.size();

        string s = strs[0];
        string t = strs[n-1];
        string str = "";
        int len = min(s.size() , t.size());
        for(int i =0 ; i< len ; i++){
            if(s[i] != t[i]) break;
            str += s[i];
        }
        return str;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        s = strs[0]
        t = strs[-1]
        ans = ""

        minlen = min(len(s) , len(t))

        for i in range(minlen):
            if s[i] != t[i]:
                break
            ans += s[i]
        return ans
